docs/2.8.0/docs/canton/usermanual/upgrading.rst

```diff
@@ -194,6 +194,64 @@ The ping command creates two contracts between the admin parties, then exercises
 Version Specific Notes
 ~~~~~~~~~~~~~~~~~~~~~~
 
+Upgrade to Release 2.8
+^^^^^^^^^^^^^^^^^^^^^^
+
+Version 2.8 extends the database schema. If you use the "migrate and start" feature, the database schema will be automatically updated.
+Otherwise, perform the manual database migration steps outlined in the :ref:`database migration steps <migrating_the_database>`.
+
+Explicit contract disclosure (Beta)
+"""""""""""""""""""""""""""""""""""
+The explicit contract disclosure feature is now in Beta and is enabled by default.
+Refer to the explicit contract disclosure :ref:`documentation <explicit-contract-disclosure>` for details.
+Note that the ``participant.ledger-api.explicit-disclosure-unsafe`` feature flag is deprecated and replaced by
+``participant.ledger-api.enable-explicit-disclosure``, set to true by default.
+
+Breaking configuration changes
+""""""""""""""""""""""""""""""
+
+**KMS wrapper-key configuration value**
+The configuration value for the KMS wrapper-key now accepts a simple string.
+Update your configuration as follows:
+
+.. code:: bash
+
+    crypto.private-key-store.encryption.wrapper-key-id = { str = "..."} # version 2.7
+    crypto.private-key-store.encryption.wrapper-key-id = "..." # version 2.8
+
+
+**Indexer Schema Migration and Cache Weight Configuration**
+Remove the following configuration lines related to the indexer and Ledger API server schema migration and cache weight:
+
+.. code:: bash
+
+    participants.participant.parameters.ledger-api-server-parameters.indexer.schema-migration-attempt-backoff
+    participants.participant.parameters.ledger-api-server-parameters.indexer.schema-migration-attempts
+    participants.participant.ledger-api.max-event-cache-weight
+    participants.participant.ledger-api.max-contract-cache-weight
+
+Breaking console commands
+"""""""""""""""""""""""""
+
+**Key Management Commands**
+The ``owner_to_key_mappings.rotate_key`` command is deprecated.
+It now expects a node reference (InstanceReferenceCommon) to avoid unwanted key rotations.
+
+Packaging
+"""""""""
+
+We have reverted the packaging change introduced in version ``2.7.0``.
+The separate distribution of ``bcprov-jdk15on-1.70.jar`` along with the Canton jar is no longer included.
+Users with Oracle JRE must explicitly add the [bouncycastle](https://www.bouncycastle.org/java.html) library to the classpath
+when running Canton.
```

Comment:
Consider linking to https://docs.daml.com/canton/usermanual/installation.html or inlining the example

Reply:
Wait are we going back on this again?

Reply:
@garyverhaegen-da Can you give context to your question/comment? As is, it is difficult for us to act on it.

Reply:
Sorry; what I meant is: is Canton 2.8 going to try to drop the embedded Bouncy Castle again, like 2.7.0 tried to do? Because that was a disaster and I'd rather know before the release hits clients.

Reply:
Uh. I probably mis-read the unreleased.md file. Invoking @thibault-da who has more context.

Reply:
> I guess it also means that the installation guide in 2.8 should not say that the Canton JAR file embeds the Bouncy Castle provider as a dependency

It does include it. It's just that if you use Oracle JVM you need to manually and expliclty add it to the classpath still, so that the JVM can verify the signature of the JAR.
We're not dropping the embedded jar again, everything stays as it is :)

Reply:
Ok, thanks. I'm relieved and reassured. 🙂